state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2092,10 +2072,28 @@ impl<T: Config<I>, I: 'static> DepositApi<T::TargetChain> for Pallet<T, I> {
 		(ChannelId, ForeignChainAddress, <T::TargetChain as Chain>::ChainBlockNumber, Self::Amount),
 		DispatchError,
 	> {
+		let swap_limits = T::SwapLimitsProvider::get_swap_request_limits();
 		if let Some(params) = &refund_params {
 			ensure!(
-				params.retry_duration <= MaxSwapRetryDurationBlocks::<T, I>::get(),
-				DispatchError::Other("Retry duration too long")
+				params.retry_duration <= swap_limits.max_swap_retry_duration_blocks,
+				DispatchError::from(Error::<T, I>::SwapRetryDurationTooLong)
+			);
+		}
+
+		if let Some(params) = &dca_params {
+			ensure!(
+				params.number_of_chunks > 0 && params.chunk_interval >= SWAP_DELAY_BLOCKS,
+				DispatchError::from(Error::<T, I>::InvalidDcaParameters)
+			);
+			let total_swap_request_duration = params
+				.number_of_chunks
+				.saturating_sub(1)
```

Comment:
What's the significance of this saturating_sub(1)?

Reply:
total_swap_request_duration = (number_of_chunks - 1) * interval + swap_delay. But i'm ignoring the initial swap delay for simplicity. So a request for 2 chunks with an interval of 5 blocks = (2-1) * 5 = 5 blocks, the gap between the first chunk and the second. I think that is correct.